### PR TITLE
Issues/95: Add CICD pipeline

### PR DESCRIPTION
**Pull request by frankinspace**
Github Issue: #95 

### Description

Transition to a python poetry project and apply CICD workflow for build and publishing artifacts.

### Overview of work done

- Switch to using [poetry](https://python-poetry.org/docs/basic-usage/) build
- Add contributing, changelog, code of conduct files
- Add CICD build for build, test, and publish

### Overview of verification done

Executed build against the `issues/95` branch multiple times: https://github.com/MAAP-Project/maap-py/actions/workflows/build-pipeline.yml?query=branch%3Aissues%2F95

Was able to test and verify
- CI build can manage version bumps
- Build can checkout code and commit version updates back to project as the MAAP-CICD app
- Project can be built and packaged as wheel
- Project is linted during build but results are currently ignored. Should be resolved by https://github.com/MAAP-Project/maap-py/issues/99
- Project tests are executed automatically but the tests currently fail. Test results are currently ignored by the build. Should be resolved by https://github.com/MAAP-Project/maap-py/issues/100
- Sonar scan has been integrated https://sonarcloud.io/project/overview?id=MAAP-Project_maap-py
  - Scan quality gate is currently failing for code coverage. Current code coverage with the tests that do work is about 60%. Sonar default ruleset has a minimum 80% coverage requirement. I'd suggest fixing the failing tests in https://github.com/MAAP-Project/maap-py/issues/100 to address this quality gate rather than change the default ruleset.

Was not able to test yet:
- Once the project is packaged it will be published to pypi.
- New github release will be created as part of publishing process

### Overview of integration done

Tested within this issue branch.

## PR checklist:

* [x] Linted (currently failing though)
* [x] Updated unit tests (currently has failing tests)
* [x] Updated changelog
* [ ] Integration testing
* [ ] Updated [documentation](https://github.com/MAAP-Project/maap-documentation)

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+_Explain how this change was integration tested. Provide screenshots or logs if appropriate. An example of this would be a local Harmony deployment._
+
+## PR checklist:
+
+* [ ] Linted
+* [ ] Updated unit tests
+* [ ] Updated changelog
+* [ ] Integration testing
+
```
Comment (sujen1412) [inline .github/pull_request_template.md:25]: Do we need to add a step for created pull request for updated documentation (if applicable) ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+request is addressing a Github issue, should match the name of the 
+Github issue. 
+
+If any performance improvements are being made, include graphs and charts.
+
+### Branches
+
+- `develop` (protected)
+    - Code actively being developed
+- `master` (protected)
```
Comment (sujen1412) [inline CONTRIBUTING.md:69]:
```suggestion
- `main` (protected)
```

**Review by sujen1412 — APPROVED** [review]

**Comment by sujen1412** [comment]
Minor changes to consider, other than that looks good.

File: .github/pull_request_template.md (hunk 1 above)
Comment (frankinspace) [reply .github/pull_request_template.md:25]: Doesn't look like this repo has docs yet. If we set that up, then I would add it to the checklist in the template

File: CONTRIBUTING.md (hunk 2 above)
Comment (frankinspace) [reply CONTRIBUTING.md:69]: The branch is currently `master`. We'd have to change this when we rename master -> main. Do you think we should do that now?

File: .github/pull_request_template.md (hunk 1 above)
Comment (sujen1412) [reply .github/pull_request_template.md:25]: Our documentation is currently in another repo. It is a more general documentation for the maap platform, but we should update those and any tutorials if making a significant change that could impact how users use this module. You're right we don't have one for this repo.

File: CONTRIBUTING.md (hunk 2 above)
Comment (sujen1412) [reply CONTRIBUTING.md:69]: Sorry, let's leave it to master. And tackle the change to main in another ticket.

File: .github/pull_request_template.md (hunk 1 above)
Comment (frankinspace) [reply .github/pull_request_template.md:26]: @sujen1412 added an item for documentation

**Review by sujen1412 — APPROVED** [review]

**Pull request merged**